File: eBiblioteka/UI/admin_ebiblioteka/lib/models/notification.dart
import 'package:json_annotation/json_annotation.dart';
import '../models/user.dart';

part 'notification.g.dart';

@JsonSerializable()
class Notif {
  //because there is pedifiend class nothification in flutter
  int? id;
  String? title;
  String? content;
  bool? isRead;
  int? userId;
  User? user;
  DateTime? createdAt;

  Notif(this.id, this.title, this.content, this.isRead, this.userId, this.user, this.createdAt);

  factory Notif.fromJson(Map<String, dynamic> json) => _$NotifFromJson(json);

  Map<String, dynamic> toJson() => _$NotifToJson(this);
}

File: eBiblioteka/eBiblioteka.Core/Entities/Notification.cs

namespace eBiblioteka.Core
{
    public class Notification:BaseEntity
    {
        public string Title { get; set; } = null!;
        public string? Content { get; set; }
        public bool IsRead { get; set; } = false;

        public int UserId { get; set; }
        public User User { get; set; } = null!;
    }
}
